rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,143 @@
+
+With some components requiring more stories, this could also result in the components appearing cluttered in the library. Similarly, the preview pane could appear cluttered by having many configurations in one frame. The stories and frames will have to be designed well to ensure this is not the perception. It may also be possible to hide the `Test` directory to keep them entirely out of view.
+
+By having configurations set up already in the stories it removes a lot of the flexibility of testing. The story code would need to be amended to change the state of various props if the required configurations are not already present in the stories. Without the knobs being visible, it may not be immediately clear to a QA Engineer which props the component has which can be amended and thus if all changes to these have been adequately demonstrated. QA Engineers would need to become more technical to enable more detailed examination and understanding of the stories code, although this could be seen as a benefit and is indeed the trend in the frontend testing world.
+
+If we run a visual regression test on a pull request from a 3rd party, they will be able to see the Applitools pass/fail test results but they will not be able to access the URL to the Applitools dashboard to view the full details unless an admin grants them temporary access. Where we are on one hand encouraging collaboration from other areas within Sage and beyond and expecting them to ultimately test their own improvements and fixes, this will limit how far they can test without our assistance. It will require us to grant them access to the Applitools dashboard or investigate and report their visual testing failures on their behalf, delaying completion of their contributions.
+
+# Alternatives
+
+There is some duplication between the Cypress regression tests and the unit tests. Indeed, the act of setting knobs to various states to set a prerequisite state and the result asserted on is already done in many of the unit tests. An exercise to remove duplicate tests in the regression suite could be carried out to speed up the run time. The recent removal of support of the ‘classic’ theme and related feature files reduced the regression suite run time by 1-1.5 hours. We could upgrade our Cypress package to a paid for service or move to the open source OSS plan which would allow the tests to be parallelised rather than them all running in one thread, reducing the run time further. All of these actions though would not reduce the regression suite run time by the required factor nor provide the visual comparison testing capability that Applitools integration would.
```

Comment:
Upgrading cypress to OSS plan is done [here](https://github.com/Sage/carbon/pull/2880) and we already mentioned about this before, so no need to mention about it in this section

Reply:
As commented above, these two streams of work have been done in parallel and that work to implement the OSS plan was not before this RFC was created/updated.